pkg/task/queue.go

```diff
@@ -53,12 +54,18 @@ func (q *Queue) Push(tsk *Task) error {
 	q.Lock()
 	defer q.Unlock()
 
+	return q.pushLocked(tsk)
+}
+
+// Pushes an item to the priority queue, without acquiring a lock
+func (q *Queue) pushLocked(tsk *Task) error {
```

Comment:
Let's either call it `unsafePush` or just `push` (and assume private functions are unsafe).